pydis_site/apps/api/viewsets/bot/whitelist.py

```diff
@@ -0,0 +1,69 @@
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework import mixins
+from rest_framework.viewsets import GenericViewSet
+
+from pydis_site.apps.api.models.bot.whitelist import Whitelist
+from pydis_site.apps.api.serializers import WhitelistSerializer
+
+
+class WhitelistViewSet(
+    mixins.CreateModelMixin,
+    mixins.ListModelMixin,
+    mixins.DestroyModelMixin,
+    GenericViewSet,
+):
+    """
+    View providing create, list and  delete operations on whitelist shown by our bot.
```

Comment:
This suggestion assumes you've added `RetrieveModelMixin` and thus made it support CRUD.

```suggestion
    View providing CRUD operations on the whitelist used by the Discord bot's filters.
```